lib/v2/nationalgeographic/latest-stories.js

```diff
@@ -0,0 +1,89 @@
+
+const findNatgeo = ($) => {
+    let json = null;
+    $('script').each(function () {
+        if (json !== null) {
+            return;
+        }
+        const text = $(this).text();
+        if (text.startsWith("window['__natgeo__']")) {
+            const match = /^window\['__natgeo__'\]=(.*);$/.exec(text);
+            if (match !== null) {
+                json = JSON.parse(match[1]);
+            }
+        }
+    });
+    return json;
+};
+
+module.exports = async (ctx) => {
+    const currentUrl = 'https://www.nationalgeographic.com/pages/topic/latest-stories';
+    const response = await got(currentUrl);
+    const $ = cheerio.load(response.data);
+    const links = [];
+    findNatgeo($).page.content.hub.frms.forEach((e) => {
+        e.mods.forEach((m) => {
+            if (m.tiles === undefined) {
+                return;
+            }
+            m.tiles.forEach((t) => {
+                if (t.ctas === undefined || t.ctas.length !== 1 || t.ctas[0].text !== 'natgeo.ctaText.read') {
+                    return;
+                }
+                links.push(t.ctas[0].url);
+            });
+        });
+    });
+    const items = await Promise.all(
+        links.map((link) =>
+            ctx.cache.tryGet(link, async () => {
+                const response = await got(link);
+                const $ = cheerio.load(response.data);
+                const article = findNatgeo($).page.content.article;
+                let bodyTile = null;
+                article.frms.forEach((f) => {
+                    if (bodyTile !== null || f.cmsType !== 'ArticleBodyFrame' || f.mods === undefined) {
+                        return;
+                    }
+                    f.mods.forEach((m) => {
+                        if (bodyTile !== null || m.edgs === undefined) {
+                            return;
+                        }
+                        m.edgs.forEach((e) => {
+                            if (bodyTile !== null || e.cmsType === undefined || e.cmsType !== 'ArticleBodyTile') {
+                                return;
+                            }
+                            bodyTile = e;
+                        });
+                    });
+                });
+                if (bodyTile === null || bodyTile.bdy === undefined) {
+                    return null;
+                }
+                const $body = cheerio.load('<body>');
+                bodyTile.bdy.forEach((b) => {
+                    if (b.type === 'p') {
+                        $body('body').append(`<p>${b.cntnt.mrkup}</p>`);
+                    }
+                });
+                return {
+                    title: article.meta.ttl,
+                    author: bodyTile.schma.athrs.map((a) => a.name).join(','),
+                    description: $body.html(),
+                    category: article.meta.pgTxnmy.sources,
+                    pubDate: parseDate(bodyTile.pbDt),
+                    link,
+                };
```

Comment:
- You can make use of [`.find()`](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Array/find)
- Use `art-template` for rendering custom HTML
```js
                const mods = findNatgeo($).page.content.article.frms.find((f) => f.cmsType === 'ArticleBodyFrame').mods;

                const bodyTile = mods.find((m) => m.edgs[0].cmsType === 'ArticleBodyTile').edgs[0];

                item.author = bodyTile.schma.athrs.map((a) => a.name).join(', ');
                item.description = art(join(__dirname, 'templates/stories.art'), {
                    ldMda: bodyTile.ldMda,
                    description: bodyTile.dscrptn,
                    body: bodyTile.bdy,
                });
                item.pubDate = parseDate(bodyTile.pbDt);
```
```mustache
{{ if ldMda }}
    <figure>
        <img src="{{ ldMda.image.src }}" alt="{{ ldMda.image.altText }}">
        <figcaption>{{ ldMda.caption }}</figcaption>
    </figure>
{{ /if }}

{{ if description }}
    <p><b>{{ dscrptn }}</b></p>
{{ /if }}

{{ each body b }}
    {{ if b.type === 'p' }}
        <p>{{@ b.cntnt.mrkup }}</p>
    {{ else if b.type === 'inline' }}
        {{ if b.cntnt.cmsType === 'image' }}
            <figure>
                <img src="{{ b.cntnt.image.src }}" alt="{{ b.cntnt.image.altText }}">
                <figcaption>{{@ b.cntnt.caption }}</figcaption>
            </figure>
        {{ else if b.cntnt.cmsType === 'imagegroup' }}
            {{ each b.cntnt.images img }}
                <figure>
                    <img src="{{ img.image.src }}" alt="{{ img.image.altText }}">
                    <figcaption>{{@ img.caption }}</figcaption>
                </figure>
            {{ /each }}
        {{ else if b.cntnt.cmsType === 'editorsNote' }}
            <p>{{@ b.cntnt.note }}</p>
        {{ else if b.cntnt.cmsType === 'listicle' }}
            <p>{{@ b.cntnt.text }}</p>
        {{ else if b.cntnt.cmsType === 'pullquote' }}
            <backquote>{{@ b.cntnt.quote }}</backquote>
        {{ else if b.cntnt.cmsType === 'source' }}
            <a href="{{ b.cntnt.src }}">{{ b.cntnt.src }}</a>
        {{ else if b.cntnt.cmsType === 'video' }}
            <figure>
                <img src="{{ b.cntnt.image.src }}" alt="{{ b.cntnt.image.altText }}">
                <figcaption>{{@ b.cntnt.title }}</figcaption>
                <figcaption>{{@ b.cntnt.description }}</figcaption>
            </figure>
        {{ /if }}
    {{ /if }}
{{ /each }}

```